RGPosadas/LitterRally
# US-7.2: Müll-Profile

This PR closes #31 

**To Test**
1. `nx serve mull-ui` and `nx serve mull-api`
2. Create a user
3. Head to http://localhost:4200/profile to see their profile
4. Click on `Edit Profile` to make further changes to your user's profile

**Notes**
- You may have seen that at first the GQL queries used fragments and I had to revert them in my frontend testing commit. This is due to the fact that MockedProvider was not able to correctly mock data that came from fragments.
- You will see `await new Promise(resolve => setTimeout(resolve, 0));` in the frontend tests, and that is the correct way to test the success state of a MockedProvider [according to the docs](https://www.apollographql.com/docs/react/development-testing/testing/#testing-the-success-state) as the default is `loading: true`.
